charms/reactive/bus.py

```diff
@@ -72,38 +77,66 @@ class states(StateList):
 
 @cmdline.subcommand()
 @cmdline.no_output
+def set_flag(flag, value=None):
+    """
+    Set the given flag as active.
+
+    :param str flag: Name of flag to set.
+    :param value: For internal use only.
+    """
+    old_flags = get_flags()
+    unitdata.kv().update({flag: value}, prefix='reactive.states.')
+    if flag not in old_flags:
+        FlagWatch.change(flag)
+
+
+@cmdline.subcommand()
+@cmdline.no_output
 def set_state(state, value=None):
+    """DEPRECATED Alias of set_flag"""
+    set_flag(state, value)
```

Comment:
Right.  I'm definitely leaning toward just dropping the decorator; I had thought it would save more duplication than it ended up actually doing.